nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EncryptAttributes.java

```diff
@@ -0,0 +1,508 @@
+            throw new ProcessException(e);
+        }
+        return encodedEncryptedStr;
+    }
+
+    private Set<String> getAttrToEncrypt(FlowFile flowFile, PropertyValue attrsToEncryptProp,
+                                         PropertyValue attrSelectRegEx, String algorithm, boolean encrypt) {
+        String attrsToEncryptPropVal = attrsToEncryptProp.getValue();
+        String regex = attrSelectRegEx.getValue();
+        Set<String> flowFileAttrs = flowFile.getAttributes().keySet();
+        Set<String> attrsToEncrypt;
+
+        if (attrsToEncryptPropVal.equals(CORE_ATTR)) {
+            attrsToEncrypt = new HashSet<>();
+        } else {
+            attrsToEncrypt = new HashSet<>(flowFileAttrs);
+        }
+
+        if (attrsToEncryptPropVal.equals(ALL_EXCEPT_CORE_ATTR)
+                || attrsToEncryptPropVal.equals(CORE_ATTR)) {
+            //traverse core attributes and add/remove as per the prop value.
+            for (CoreAttributes attr : CoreAttributes.values()) {
+                if (flowFileAttrs.contains(attr.key()) && attrsToEncryptPropVal.equals(ALL_EXCEPT_CORE_ATTR)) {
+                    attrsToEncrypt.remove(attr.key());
+                } else if (flowFileAttrs.contains(attr.key()) && attrsToEncryptPropVal.equals(CORE_ATTR)) {
+                    attrsToEncrypt.add(attr.key());
+                }
+            }
+        }
+
+        if (attrsToEncryptPropVal.equals(CUSTOM_ATTR)) {
+
+            //get list of all the attributes matching regex
+            if (regex != null && !regex.equals(".*")) {
+                attrsToEncrypt.clear();
+                Pattern pattern = Pattern.compile(regex);
+                for (String str : flowFileAttrs) {
+                    if (pattern.matcher(str).matches()) {
+                        attrsToEncrypt.add(str);
+                    }
+                }
+            }
+
+            //check if property-key is present in attrsToEncrypt and if expression-lang condition
+            //return true then encrypt/decrypt it.
+            if (!propMap.isEmpty()) {
+                HashSet<String> attrsToEncryptClone = new HashSet<>(attrsToEncrypt);
+                for(String attr: attrsToEncryptClone) {
+                    if (propMap.containsKey(attr)) {
+                        boolean matches = propMap.get(attr).evaluateAttributeExpressions(flowFile).asBoolean();
+                        if (!matches){
+                            attrsToEncrypt.remove(attr);
+                            getLogger().warn("{} expression-language expression evaluates to false",
+                                    new Object[]{propMap.get(attr).getValue()});
+                        }
+                    } else {
+                        attrsToEncrypt.remove(attr);
+                    }
+                }
+
+            }
+        }
+
+        attrsToEncrypt.remove(CoreAttributes.UUID.key());
+        if (EncryptProcessorUtils.isPGPAlgorithm(algorithm)) {
```

Comment:
Please explain why you are doing this. The PGP keyring filename is not the same as the filename attribute on the flowfile.